Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,153 @@
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+
+class UIKitUserInteractions {
+    fileprivate static let amplitudeInstances = NSHashTable<Amplitude>.weakObjects()
+
+    private static let queue = DispatchQueue(label: "com.amplitude.autocapture", target: .global(qos: .utility))
+
+    private static let initializeSwizzle: () = {
+        swizzleSendAction()
+    }()
+
+    static func register(_ amplitude: Amplitude) {
+        queue.sync {
+            amplitudeInstances.add(amplitude)
+        }
+        initializeSwizzle
+    }
+
+    private static func swizzleSendAction() {
+        let applicationClass = UIApplication.self
```

Comment:
does this work in app extensions (share extensions would be good to test)?  iirc uiapplication.shared support can be iffy, not sure how this would effect swizzling, though.

Reply:
Another approach might be to swizzle the sendAction method of UIControl instead. Does this make any difference in terms of support for app extensions?

Reply:
It would be nice to test this in a quick sample app - it's a different enough scenario that I'm not sure what else may break.  Otherwise, UIControl.sendAction should be extension safe.

Reply:
UIApplication.sendAction seems to be safe in app extensions!